pytest_httpserver/httpserver.py

```diff
@@ -331,6 +372,10 @@ def __init__(self, host=DEFAULT_LISTEN_HOST, port=DEFAULT_LISTEN_PORT, ssl_conte
         self.handlers = RequestHandlerList()
         self.permanently_failed = False
         self.ssl_context = ssl_context
+        self.default_waiting_settings = (
+            default_waiting_settings if default_waiting_settings is not None else WaitingSettings())
+        self._waiting_settings = copy(self.default_waiting_settings)
+        self._waiting_result = queue.LifoQueue(maxsize=1)
```

Comment:
Have you tried this with threading events? It is clear that currently you want to differentiate between the outcome of the test running but that could be also done by examining the state of this object when the event is fired.
(this is a question only)

Reply:
No, I have not. But if I use threading events I need an additional variable for storing the result. Also, this code will not be thread-safe. I think using queues is the better option.